pom.xml

```diff
@@ -1154,131 +1170,128 @@
-                                    if (relativePath.matches(attachmentFilepathPattern)) {
-                                        attachmentPathByFile.put(path.getFileName().toString(), path);
-                                    }
-                                }
-                            }
-                        });
-                    } catch (Exception error) {
-                        // Supplement diagnostics
-                        error.printStackTrace(System.err);
-                        throw error;
-                    } finally {
-                        paths.close();
-                    }
-
-                    // Check if no attachments were found
-                    if (${attachmentCount} != attachmentPathByFile.size()) {
-                        System.err.println("Attachments:");
-                        int[] i = {0};
-                        attachmentPathByFile.values().forEach(new Consumer() {
-                            public void accept(Path path) {
-                                System.err.format("  [%2d] %s%n", new Object[]{++i[0], path});
-                            }
-                        });
-                        System.err.format(
-                                "Error: Was expecting %d attachments, found %d!%n",
-                                new Object[]{${attachmentCount}, attachmentPathByFile.size()});
-                        System.err.println("Tip: Have you already executed the Maven `package` goal?");
-                        System.exit(1);
-                    }
-
-                    // Create the binary distribution
-                    pathByFile.putAll(attachmentPathByFile);
-                    zip("target/bin.zip", pathByFile);
-                    System.out.format("Generated the binary distribution (`bin.zip`) containing following %d files:%n", new Object[] {pathByFile.size()});
-                    for (String file : pathByFile.keySet()) {
-                        System.out.println("-> " + file);
-                    }]]></script>
+                  <scripts>
+                    <script><![CDATA[import java.io.*;
+                      import java.nio.file.*;
+                      import java.util.*;
+                      import java.util.function.*;
+                      import java.util.regex.*;
+                      import java.util.stream.*;
+                      import java.util.zip.*;
+                      import java.time.Instant;
+
+                      import org.eclipse.jgit.dircache.*;
+                      import org.eclipse.jgit.lib.Repository;
+                      import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+
+                      long timestampMillis = Instant.parse(project.properties['project.build.outputTimestamp']).toEpochMilli()
+                      def getProperty = { String name ->
+                          return session.userProperties[name] ?: session.systemProperties[name]
+                      }
```

Comment:
Curious: What was wrong with hardcoding Maven properties?

Reply:
I prefer not to inject the value of a property, since it can break the script. For example `"${attachmentFilepathPattern}"` will not compile if the value is `.*\.jar`.

I refactored the script to bind all the necessary string parameters in https://github.com/apache/logging-parent/pull/241/commits/978499680344d8b77833bcc6c4a70bd7650f0397.